components/PlayButton.tsx

```diff
@@ -0,0 +1,11 @@
+import { PauseIcon, PlayIcon } from "../icons/PlayIcons";
+
+const PlayButton = ({ hook, media }) => {
+  const [playing, toggle] = hook(media);
```

Comment:
I think it would be cleaner to have an `AudioPlayButton` and a `VideoPlayButton`. media can be a `string` or a `ref` in this scenario, which can be a bit odd. 

I wasn't sure how to express why passing a hook as a prop is wrong, this answer is spot on. 

<img width="457" alt="Screen Shot 2022-07-29 at 5 30 25 PM" src="https://user-images.githubusercontent.com/11382084/181803874-cb45c9d2-a58c-4f43-8654-1c94e87c3016.png">

Reply:
Basically, having a hook as a prop invites a developer to dynamically change the hook that gets passed in, which will cause a bug. Look up react "rules of hooks"